Recipes/InheritedTypeJsonConverter/InheritedTypeJsonConverter.cs

```diff
@@ -335,7 +376,33 @@ public override bool CanWrite
         /// <inheritdoc />
         public override bool CanConvert(Type objectType)
         {
-            return false;
+            // WriteJson needs to use the JsonSerializer passed to the method so that
+            // the various settings in the serializer are utilized for writing.
+            // Using the serializer as-is, however, will cause infinite recursion because
+            // this Converter is utilized by the serializer.  If we modify the serializer
+            // to remove this converter, it will affect all other consumers of the serializer.
+            // Also, the object to serialize might contain types that require this converter.
+            // The only way to manage this to store some state when WriteJson is called,
+            // detect that state here, and tell json.net that we cannot convert the type.
+            if (WriteJsonCalled.Value)
+            {
+                WriteJsonCalled.Value = false;
+                return false;
+            }
+
+            var bindableAttribute = GetBindableAttribute(objectType);
+            if (bindableAttribute == null)
+            {
+                return false;
+            }
+
+            // Two-way instructs the converter to add the name of the type to the JSON payload
+            // when serializing.  This is sometimes required to disambiguate types when de-serializing.
+            // Sometimes, multiple types have the same named properties and share an abstract parent.  De-serializing
+            // the JSON into the abstract parent won't work without some extra information to determine which
+            // of those multiple types to create.  In this case, we want to call our implementation of WriteJson.
+            var isTwoWayBindable = IsTwoWayBindable(bindableAttribute);
+            return isTwoWayBindable;
```

Comment:
Am I missing something about how bindable, but not two-way types are written? Seems this method only returns true if Both Bindable and TwoWay.